bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java

```diff
@@ -0,0 +1,344 @@
+            case CURRENT_L3:
+                return new QuantityType<>((Double) (goeResponse.energy[6] / 10d), SmartHomeUnits.AMPERE);
+            case POWER_L1:
+                return new QuantityType<>((Double) (goeResponse.energy[7] / 10d), SmartHomeUnits.WATT);
+            case POWER_L2:
+                return new QuantityType<>((Double) (goeResponse.energy[8] / 10d), SmartHomeUnits.WATT);
+            case POWER_L3:
+                return new QuantityType<>((Double) (goeResponse.energy[9] / 10d), SmartHomeUnits.WATT);
+        }
+        return UnDefType.UNDEF;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            // we can not update single channels and refresh is triggered automatically
+            // anyways
+            return;
+        }
+
+        String key = null;
+        String value = null;
+        switch (channelUID.getId()) {
+            case MAX_AMPERE:
+                key = "amp";
+                if (command instanceof DecimalType) {
+                    value = String.valueOf(((DecimalType) command).intValue());
+                } else if (command instanceof QuantityType<?>) {
+                    value = String.valueOf(
+                            ((QuantityType<ElectricCurrent>) command).toUnit(SmartHomeUnits.AMPERE).intValue());
+                }
+                break;
+            case SESSION_CHARGE_CONSUMPTION_LIMIT:
+                key = "dwo";
+                if (command instanceof DecimalType) {
+                    value = String.valueOf(((DecimalType) command).intValue() * 10);
+                } else if (command instanceof QuantityType<?>) {
+                    value = String.valueOf(
+                            ((QuantityType<Energy>) command).toUnit(SmartHomeUnits.KILOWATT_HOUR).intValue() * 10);
+                }
+                break;
+            case ALLOW_CHARGING:
+                key = "alw";
+                if (command instanceof OnOffType) {
+                    value = command == OnOffType.ON ? "1" : "0";
+                }
+                break;
+            default:
+        }
+        if (key != null && value != null) {
+            sendData(key, value);
+        } else {
+            logger.warn("Could not update channel {} with key {} and value {}", channelUID.getId(), key, value);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(GoEChargerConfiguration.class);
+        allChannels = getThing().getChannels().stream().map(channel -> channel.getUID().getId())
+                .collect(Collectors.toList());
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        startAutomaticRefresh();
+        logger.debug("Finished initializing!");
+    }
+
+    private String getUrl(String type) {
+        String urlStr = type;
+        return urlStr.replace("%IP%", StringUtils.trimToEmpty(config.ip));
```

Comment:
You can use the `type` parameter straight away instead of first assigning it to `urlStr` or rename it from `type` to `urlStr`.

```suggestion
        return type.replace("%IP%", StringUtils.trimToEmpty(config.ip));
```

Reply:
Good catch, i think i did something else before and in the end forget to remove it when i did not need it anymore.